source/s2n/s2n_tls_channel_handler.c

```diff
@@ -924,6 +950,48 @@ static int s2n_monotonic_clock_time_nanoseconds(void *context, uint64_t *time_in
     return 0;
 }
 
+static int s_tls_ctx_pkcs11_setup(struct s2n_ctx *s2n_ctx, const struct aws_tls_ctx_options *options) {
+    /* anything initialized in this function is cleaned up during s_s2n_ctx_destroy()
+     * so don't worry about cleaning up unless it's some tmp heap allocation */
+
```

Comment:
Is there a need to sanitize options here, to make sure it has everything we need.

Reply:
nvm, I see the validation later.

Reply:
check out the changes in`source/tls_channel_handler.c`

they're sanitized in the platform-agnostic code that copies the data from `aws_tls_ctx_pkcs11_options` into `aws_tls_ctx_options`

I'll add a comment to that effect